host/sgx/enter.c

```diff
@@ -139,19 +121,31 @@ int __oe_host_stack_bridge(
 
 /**
  * Setup the ecall_context.
+ * This function is assumed to only be called by oe_enter and
+ * should never be inline. Doing so allows for recording the
+ * stack frame as the caller to the oe_enter. The stack frame
+ * information is used to stitch the stack upon an enclave entry.
  */
-OE_INLINE void _setup_ecall_context(oe_ecall_context_t* ecall_context)
+OE_NEVER_INLINE
+static void _setup_ecall_context(oe_ecall_context_t* ecall_context)
 {
     oe_thread_binding_t* binding = oe_get_thread_binding();
+    uint64_t* frame = __builtin_frame_address(0);
```

Comment:
This needs to be `__builtin_frame_address(1)` in order to fetch the frame of the caller. The 0th frame is that of the current function. It becomes invalid when the function returns.